migtests/scripts/live-migration-fallf-run-test.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+set -e
+set -m
+
+if [ $# -ne 1 ]
+then
+	echo "Usage: $0 TEST_NAME"
+	exit 1
+fi
+
+set -x
+
+export TEST_NAME=$1
+
+export REPO_ROOT="${PWD}"
+export SCRIPTS="${REPO_ROOT}/migtests/scripts"
+export TESTS_DIR="${REPO_ROOT}/migtests/tests"
+export TEST_DIR="${TESTS_DIR}/${TEST_NAME}"
+export EXPORT_DIR=${EXPORT_DIR:-"${TEST_DIR}/export-dir"}
+
+export PYTHONPATH="${REPO_ROOT}/migtests/lib"
+
+# Order of env.sh import matters.
+source ${TEST_DIR}/env.sh
+
+if [ "${SOURCE_DB_TYPE}" = "oracle" ]
+then
+	source ${SCRIPTS}/${SOURCE_DB_TYPE}/live_env.sh 
+	source ${SCRIPTS}/${SOURCE_DB_TYPE}/ff_env.sh
+else
+	source ${SCRIPTS}/${SOURCE_DB_TYPE}/env.sh
+fi
+
+source ${SCRIPTS}/yugabytedb/env.sh
+
+source ${SCRIPTS}/functions.sh
+
+main() {
+
+	echo "Deleting the parent export-dir present in the test directory"
+	rm -rf ${EXPORT_DIR}	
+	echo "Creating export-dir in the parent test directory"
+	mkdir -p ${EXPORT_DIR}
+	echo "Assigning permissions to the export-dir to execute init-db, cleanup-db scripts"
+	chmod +x ${TEST_DIR}/init-db ${TEST_DIR}/cleanup-db
+	
+	step "START: ${TEST_NAME}"
+	print_env
+
+	pushd ${TEST_DIR}
+
+	step "Prepare FF metadata tables"
+	run_sqlplus_as_sys ${FF_DB_NAME} ${SCRIPTS}/oracle/fall-forward-prep.sql
+
+	step "Initialise source and fall forward database."
+	./init-db
+
+	step "Grant source database user permissions"
+	if [ "${SOURCE_DB_TYPE}" = "oracle" ]
+	then
+		grant_permissions_for_live_migration_oracle ${ORACLE_CDB_NAME} ${SOURCE_DB_NAME}
+	else
+		grant_permissions ${SOURCE_DB_NAME} ${SOURCE_DB_TYPE} ${SOURCE_DB_SCHEMA}
+	fi
+
+	step "Check the Voyager version installed"
+	yb-voyager version
+
+	step "Export schema."
+	export_schema
+	find ${EXPORT_DIR}/schema -name '*.sql' -printf "'%p'\n"| xargs grep -wh CREATE
+
+	step "Analyze schema."
+	analyze_schema
+	tail -20 ${EXPORT_DIR}/reports/report.txt
+
+	step "Fix schema."
+	if [ -x "${TEST_DIR}/fix-schema" ]
+	then
+		 "${TEST_DIR}/fix-schema"
+	fi
+
+	step "Analyze schema."
+	analyze_schema
+	tail -20 ${EXPORT_DIR}/reports/report.txt
+
+	step "Create target database."
+	run_ysql yugabyte "DROP DATABASE IF EXISTS ${TARGET_DB_NAME};"
+	run_ysql yugabyte "CREATE DATABASE ${TARGET_DB_NAME}"
+
+	step "Import schema."
+	import_schema
+	run_ysql ${TARGET_DB_NAME} "\dt"
+
+	step "Export data."
+	# false if exit code of export_data is non-zero
+	export_data --export-type "snapshot-and-changes" || { 
+		tail_log_file "yb-voyager-export-data.log"
+		tail_log_file "debezium-source_db_exporter.log"
+		exit 1
+	} &
+
+	# Storing the pid for the export data command
+	exp_pid=$!
+
+	# Killing the export process in case of failure
+	trap "kill_process -${exp_pid} ; exit 1" SIGINT SIGTERM EXIT SIGSEGV SIGHUP
+
+	# Waiting for snapshot to complete
+	timeout 100 bash -c -- 'while [ ! -f ${EXPORT_DIR}/metainfo/flags/exportDataDone ]; do sleep 3; done'
```

Comment:
I don't think this is required anymore right? now we wait in import data/ ff setup itself for this thing

Reply:
Yes correct. Sanyam has removed it in his PR.